zavod/zavod/shed/enforcements.py

```diff
@@ -0,0 +1,18 @@
+from datetime import datetime
+
+from zavod import helpers as h
+from zavod.context import Context
+
+
+MAX_ENFORCEMENT_DAYS = 365 * 5
+
+
+def within_max_age(
+    context: Context,
+    date: datetime | str,
+    max_age_days: int = MAX_ENFORCEMENT_DAYS,
+) -> bool:
+    if isinstance(date, str):
+        date = date.strip()
+    date = h.extract_date(context.dataset, date)[0]
```

Comment:
we need to registry.date.clean this to be sure it's not just the unparsable date spat back out

Reply:
Should `extract_date` have an argument to not do that?

Reply:
That's a lot nicer